File: android/app/src/main/java/com/zhuodazi/android/DeskPetApi.java
package com.zhuodazi.android;

final class DeskPetApi {
    static final String HOST = "in.desktoppet.online";
    static final String BASE_URL = "https://" + HOST;
    static final String SIGNING_PUBLIC_KEY_SPKI =
        "MCowBQYDK2VwAyEANjBEMMQ5TY+0ECNoRqQy9780eoVOzkKpzFDq2TwLytU=";

    static final String ACTIVATE = "/api/activate";
    static final String TRIAL = "/api/trial";
    static final String FEEDBACK = "/api/feedback";
    static final String INTERACTION_PROFILE = "/api/interactions/profile";
    static final String INTERACTION_EVENTS = "/api/interactions/events";
    static final String CONTENT_BATCH = "/api/content/batch";
    static final String CONTENT_OFFLINE_PACK = "/api/content/offline-pack";
    static final String COMPANION = "/api/companion";
    static final String COMPANION_PAIR = "/api/companion/pair";
    static final String COMPANION_DELIVERIES = "/api/companion/deliveries";
    static final String COMPANION_HALL = "/api/companion/hall";
    static final String COMPANION_HALL_DELIVERIES = "/api/companion/hall/deliveries";
    static final String TRIAL_VISIT_PLAY = "/api/trial/visit-stickers/play";
    static final String ANALYTICS_EVENTS = "/api/analytics/events";
    static final String COMPANION_DOWNLOAD_PREFIX = "/api/companion/";
    static final String SITE_SETTINGS = "/api/public/site-settings";
    static final String UPDATE_LATEST = "/api/update/latest";
    static final String DOWNLOAD_PATH_PREFIX = "/downloads/";

    private DeskPetApi() { }
}

File: macos/Sources/ZhuoDaziMac/DeskPetApi.swift
import Foundation

enum DeskPetApi {
    static let host = "in.desktoppet.online"
    static let baseURL = URL(string: "https://\(host)")!
    static let signingPublicKeySPKI = "MCowBQYDK2VwAyEANjBEMMQ5TY+0ECNoRqQy9780eoVOzkKpzFDq2TwLytU="
    static let downloadPathPrefix = "/downloads/"

    static let activate = baseURL.appendingPathComponent("api/activate")
    static let trial = baseURL.appendingPathComponent("api/trial")
    static let feedback = baseURL.appendingPathComponent("api/feedback")
    static let interactionProfile = baseURL.appendingPathComponent("api/interactions/profile")
    static let interactionEvents = baseURL.appendingPathComponent("api/interactions/events")
    static let contentBatch = baseURL.appendingPathComponent("api/content/batch")
    static let contentOfflinePack = baseURL.appendingPathComponent("api/content/offline-pack")
    static let companion = baseURL.appendingPathComponent("api/companion")
    static let companionPair = companion.appendingPathComponent("pair")
    static let companionDeliveries = companion.appendingPathComponent("deliveries")
    static let companionHall = companion.appendingPathComponent("hall")
    static let companionHallDeliveries = companionHall.appendingPathComponent("deliveries")
    static let trialVisitPlay = baseURL.appendingPathComponent("api/trial/visit-stickers/play")
    static let analyticsEvents = baseURL.appendingPathComponent("api/analytics/events")
    static let updateLatest = baseURL.appendingPathComponent("api/update/latest")
    static let siteSettings = baseURL.appendingPathComponent("api/public/site-settings")
}
